components/gitpod-protocol/src/protocol.ts

```diff
@@ -151,18 +151,29 @@ export interface UserFeatureSettings {
 export const WorkspaceFeatureFlags = { "full_workspace_backup": undefined, "fixed_resources": undefined };
 export type NamedWorkspaceFeatureFlag = keyof (typeof WorkspaceFeatureFlags);
 
-export interface UserEnvVarValue {
-    id?: string;
+export interface EnvVarWithValue {
     name: string;
-    repositoryPattern: string;
     value: string;
 }
+
+export interface ProjectEnvVarWithValue extends EnvVarWithValue {
+    id: string;
+    projectId: string;
+}
+
+export type ProjectEnvVar = Omit<ProjectEnvVarWithValue, 'value'>;
+
+export interface UserEnvVarValue extends EnvVarWithValue {
+    id?: string;
+    repositoryPattern: string; // DEPRECATED: Use ProjectEnvVar instead - https://github.com/gitpod-com/gitpod/issues/5322
+}
 export interface UserEnvVar extends UserEnvVarValue {
     id: string;
     userId: string;
     deleted?: boolean;
 }
 
+// DEPRECATED: Use ProjectEnvVar instead - https://github.com/gitpod-com/gitpod/issues/5322
```

Comment:
This is a bit unexpected. I think I agree on the `repositoryPattern`. But user env-vars are still useful for user-wide stuff, right?

Reply:
Agree, maybe I was a bit heavy-handed here. My point of view is that UserEnvVar is 98% `repositoryPattern` and 2% just `name` and `value`, hence we're deprecating most of UserEnvVar.

Also, I'm not quite sure what a User-wide variable concept looks like with `repositoryPattern` removed. E.g., maybe you'll want a few variables _everywhere_ (e.g. `EDITOR=vim`), but for personal keys, I think it makes more sense to have "User-specific Project variables" (i.e. I go to the Gitpod project, and it shows "Project variables for everyone in the Team", as well as "Project variables just for my User").

That's sort of why I wanted to mark UserEnvVar as deprecated until it has been refactored, but I'm happy to mark just `repositoryPattern` for now.

Reply:
> User-wide variable concept

Hm. IMO the `repositoryPattern` still offers a way of flexibility I don't get otherwise, no?
If there's a service that I want to use in the majority of my projects, and I have a personal access token for: Do you see that on project level? I had a user-level view with repo selection (like GitHub) in mind.

But anyway, I'd be in favor of deprecating `repositoryPattern` only for now, and sort the rest out later.

Reply:
> If there's a service that I want to use in the majority of my projects, and I have a personal access token for: Do you see that on project level?

Good point, this use case doesn't seem to make sense on Project level.

> I had a user-level view with repo selection (like GitHub) in mind.

Precise repository selection sounds like a good option (and isn't affected by https://github.com/gitpod-com/gitpod/issues/5322 like `repositoryPattern`).

However, I wonder how common/important this particular use case is (i.e. whether we'll want to implement a full User-wide multi-reposelection per variable, or if we can live without this feature). In any case, we don't need to make this decision now. 😇

Reply:
Made the deprecation warnings more specific to `repositoryPattern`. ✅ Thanks for the feedback!